src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -49,15 +60,65 @@ public static string EmitVerb(Endpoint endpoint)
      * their validity (optionality), invoke the underlying handler with
      * the arguments bound from HTTP context, and write out the response.
      */
-    public static string EmitRequestHandler()
+    public static string EmitRequestHandler(this Endpoint endpoint)
     {
-        return """
-Task RequestHandler(HttpContext httpContext)
-                {
-                        var result = handler();
-                        return httpContext.Response.WriteAsync(result);
-                }
-""";
+        var code = new CodeWriter(new StringBuilder());
+        code.Indent(5);
```

Comment:
When @captainsafia was getting @brunolins16 up to speed yesterday we were discussing the ```CodeWriter``` in general. My feedback was that using the CodeWriter rather than simple string concatenation/interpolation actually made it harder to reason over the generated code. I'd rather see us drop the CodeWriter (at the expense of indentation) and then either do some code formatting at the end before handing it back to Roslyn or just leave it a little messy. We can also do a Format Document on the generated code after build if we need to read it.

Reply:
The Regex source generator uses `IndentedTextWriter`. Seems to work really well there. You just `writer.Indent++` and `--` when you are done.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs#L30